### PR TITLE
darwin bootstrapping: Avoid overriding aliases for LLVM

### PR DESCRIPTION
**Pull request by Ericson2314**
###### Motivation for this change

Respect the fix points and aliases by overriding originals.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/stdenv/darwin/default.nix [hunk 1]
```diff
@@ -272,13 +276,18 @@ in rec {
...
-       };
+      llvmPackages_5 = super.llvmPackages_5 // (let
+        tools = super.llvmPackages_5.tools.extend (_: _: {
+          llvm = llvmPackages_5.llvm.override { inherit libcxxabi; };
+          clang-unwrapped = llvmPackages_5.clang-unwrapped.override { llvm = self.llvmPackages_5.llvm; };
+        });
+        libraries = super.llvmPackages_5.libraries.extend (_: _: {
+          inherit (llvmPackages_5) libcxx libcxxabi;
+        });
+      in { inherit tools libraries; } // tools // libraries);
```
Comment (Ericson2314) [inline pkgs/stdenv/darwin/default.nix]: @matthewbauer at some point lets make the `// tools //libraries` happen in `aliases.nix` so the overrides here can be simpler.

**Comment by Ericson2314** [comment]
Pretty sure the slight Darwin rebuild is due to referees persisted that weren't before.

File: pkgs/stdenv/darwin/default.nix (hunk 1 above)
Comment (LnL7) [reply pkgs/stdenv/darwin/default.nix]: Yeah, or just deprecate the versioned aliases and only expose attributes for the current version.

Comment (Ericson2314) [reply pkgs/stdenv/darwin/default.nix]: Good idea!

**Comment by Ericson2314** [comment]
Fixed with https://github.com/NixOS/nixpkgs/pull/42174

**Pull request merged**